src/overlay_94.c

```diff
@@ -0,0 +1,228 @@
+            ov94_021E5BA0(unkA);
+            unkA->unk0++;
+            break;
+        case 5:
+            unkA->unk4++;
+            if(unkA->unk4 == unkA->unk8) {
+                sub_0207ECE0(unkPtr, unkPtr->partyMonIndex);
+            }
+            ov94_021E5C60();
+            if(unkA->unk4 > unkA->unk8) {
+                if(sub_020154B0(unkA->unk18) == 0) {
+                    unkA->unk0++;
+                }
+            }
+            break;
+        case 6:
+            ov94_021E5B30(unkPtr);
+            unkA->unk0++;
+            break;
+        case 7:
+            sub_020720D4(pokemon);
+            unkA->unk0++;
+            break;
+        case 8:
+            if(IsCryFinished() == FALSE) {
+                unkA->unk0++;
+            }
+            break;
+        case 9:
+            STRING* str = NewString_ReadMsgData(unkPtr->msgData, 0xbc);
+            BufferBoxMonNickname(unkPtr->unk7c4, 0, Mon_GetBoxMon(pokemon));
+            StringExpandPlaceholders(unkPtr->unk7c4, unkPtr->unk7c8, str);
+            String_dtor(str);
+            sub_0207DAEC(unkPtr, ~0, 1);
+            unkA->unk0++;
+            break;
+        case 10:
+            if (sub_02020094(unkPtr->unkc64) == 0) {
+                ov94_021E5AEC(unkPtr);
+                unkPtr->unk654->unk27 = 0;
+                return TRUE;
+            }
+            break;
+    }
+    return FALSE;
+}
+
+void ov94_021E5AEC(struct UnkStruct_Overlay_94_A* unkPtr) {
+    FreeToHeap(unkPtr->unkc80);
+    unkPtr->unkc80 = 0;
+}
+
+void ov94_021E5B04(struct UnkStruct_Overlay_94_A* unkPtr) {
+    sub_0207991C(unkPtr, 0);
+    ov94_021E5B54(unkPtr->unkc80);
+    G2x_SetBlendAlpha_(0x04000050, 0, 0x3f, 0x1f, 0);
+}
+
+void ov94_021E5B30(struct UnkStruct_Overlay_94_A* unkPtr) {
+    ov94_021E5C84(unkPtr->unkc80);
+    sub_0207991C(unkPtr, 1);
+    u16* unkPtr2 = (u16*) 0x04000050;
+    unkPtr2[0] = 0;
+}
+
+void ov94_021E5B54(struct UnkStruct_Overlay_94_C* unkPtr) {
+    sub_02014DA0();
+    void* unkPtrB = AllocFromHeap(0xc, 0x4800);
+
+    unkPtr->unk18 = sub_02014DB4(ov94_021E5CA0, ov94_021E5CC4, unkPtrB, 0x4800, 1, 0xc);
+
+    sub_02023240(0x1000, 0x384000, sub_02015524());
+}
+
+void ov94_021E5BA0(struct UnkStruct_Overlay_94_C* unkPtr) { 
+    sub_0201526C(unkPtr->unk18, sub_02015264(0xd0, unkPtr->unk10, 0xc), 0xa, 1);
+
+    switch (unkPtr->species) {
+        case SPECIES_GIRATINA: 
+            sub_02015494(unkPtr->unk18, 0, ov94_021E5C28, unkPtr);
+            sub_02015494(unkPtr->unk18, 1, ov94_021E5C28, unkPtr);
+            sub_02015494(unkPtr->unk18, 2, ov94_021E5C28, unkPtr);
+            PlaySE(SEQ_SE_PARTY_GIRATINA_FORME_CHANGE);
+            return;
+        case SPECIES_SHAYMIN: 
+            sub_02015494(unkPtr->unk18, 0, ov94_021E5C28, unkPtr);
+            sub_02015494(unkPtr->unk18, 1, ov94_021E5C28, unkPtr);
+            PlaySE(SEQ_SE_PARTY_SHAYMIN_FORME_CHANGE);
+            return;
+    }
+}
+
+/*
+static u32 _021E5CE8[] = {
+	0x8C, 0xBF, 0xFF, 0xFF
```

Comment:
This will compile to `8C 00 00 00 BF 00 00 00 FF 00 00 00 FF 00 00 00` which is incorrect.